central/detection/service/service_impl.go

```diff
@@ -340,11 +351,41 @@ func (s *serviceImpl) DetectDeployTimeFromYAML(ctx context.Context, req *apiV1.D
 		eCtx.ClusterID = clusterID
 	}
 
+	var deployments []*storage.Deployment
+	var errs []error
+
 	var runs []*apiV1.DeployDetectionResponse_Run
 	for _, r := range resources {
-		run, err := s.runDeployTimeDetect(ctx, eCtx, r, req.GetPolicyCategories())
+		d, err := s.convertK8sResource(r)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		deployments = append(deployments, d)
+	}
+	if len(deployments) == 0 && len(errs) > 0 {
+		msg := fmt.Sprintf("every deployment YAML failed to parse - aborting. Errors:\n%v", errs)
+		return nil, errox.InvalidArgs.New(msg)
```

Comment:
I am not sure about the `\n` in log messages. It may make parsing difficult. Is there another way to concatenate errors in one line and make them readable?
```suggestion
		return nil, errox.InvalidArgs.Newf("all deployment YAML failed to parse - aborting. Errors:\n%v", errs)
```